### PR TITLE
Types

### PR DESCRIPTION
**Pull request by Stefanwullems**
- QueryBuilder can be inferred
- RecordsProps get removed from components Props type
- Wrapped component doesn't have to be a class component anymore

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
 import * as React from 'react';
```
Comment (hongaar) [inline index.d.ts]: Would it be possible to change this to `import React, { ComponentType, ComponentClass } from 'react';` and get rid of the alias types?

**Comment by hongaar** [comment]
Thanks for this @Stefanwullems, will review soon.

File: index.d.ts [hunk 2]
```diff
@@ -22,17 +24,34 @@ export type WithDataProps =
...
+  | RecordsToProps<keyof TRecordProps>
+  | MapRecordsToPropsFn<TRecordProps, TOwnProps>
 
 export class DataProvider extends React.Component<DataProviderProps> {}
 
+export type ConnectedComponentClass<C, P> = ComponentClass<JSX.LibraryManagedAttributes<C, P>> & {
+  WrappedComponent: C;
+};
+
+export interface InferableComponentEnhancerWithProps<TInjectedProps, TNeedsProps> {
```
Comment (hongaar) [inline index.d.ts]: Can't really figure it out by myself... ;)
What does this interface do?

**Comment by Stefanwullems** [comment]
The any is very probably because I had to add the react types tot the devDependencies. Try running yarn and see if it changes anything.

File: index.d.ts (hunk 2 above)
Comment (Stefanwullems) [reply index.d.ts]: I copy pasted from react-redux. It infers the props of the component and the props that it is going to inject and removes the props that it injects from the props interface. Next to that it makes sure that the types of things like displayName and propTypes don’t get lost.

**Comment by hongaar** [comment]
> The any is very probably because I had to add the react types tot the devDependencies. Try running yarn and see if it changes anything.

My bad, that fixed it! 👍

**Pull request merged**